crates/query-engine/translation/src/translation/query/fields.rs

```diff
@@ -283,29 +283,80 @@ fn translate_nested_field(
+            TableSource::Collection(collection_name) => {
+                let source = TableSource::NestedField {
+                    collection_name: collection_name.clone(),
+                    type_name: nested_field_type_name,
+                    field_path: FieldPath(vec![current_column_name.clone()]),
+                };
+                let nested_field_binding_alias = state.make_table_alias(source.name_for_alias());
+                (
+                    TableSourceAndReference {
+                        source,
+                        reference: sql::ast::TableReference::AliasedTable(
+                            nested_field_binding_alias.clone(),
+                        ),
+                    },
+                    nested_field_binding_alias,
+                )
+            }
+            TableSource::NestedField {
+                collection_name,
+                type_name: _,
+                field_path,
+            } => {
+                let mut field_path = field_path.0.clone();
+                field_path.push(current_column_name.clone());
+                let source = TableSource::NestedField {
+                    collection_name: collection_name.clone(),
+                    type_name: nested_field_type_name,
+                    field_path: FieldPath(field_path),
+                };
+                let nested_field_binding_alias = state.make_table_alias(source.name_for_alias());
+                (
+                    TableSourceAndReference {
+                        source,
+                        reference: sql::ast::TableReference::AliasedTable(
+                            nested_field_binding_alias.clone(),
+                        ),
+                    },
+                    nested_field_binding_alias,
+                )
+            }
+        }
+    };
     // The FROM-clause to use for the next layer of fields returned by `translate_fields` below,
     // which brings each nested field into scope as separate columns in a sub query.
-    let nested_field_binding_alias = state.make_table_alias("nested_field_binding".to_string());
     let nested_field_from = sql::ast::From::Select {
         select: Box::new(sql::helpers::select_composite(field_binding_expression)),
-        alias: nested_field_binding_alias.clone(),
+        alias: nested_field_binding_alias,
     };
 
-    // The recursive call to the next layer of fields
-    let nested_field_table_reference = TableNameAndReference {
-        name: nested_field_type_name.as_str().into(),
-        reference: sql::ast::TableReference::AliasedTable(nested_field_binding_alias),
-    };
+    // join aliases
+    let mut join_relationship_fields: Vec<relationships::JoinFieldInfo> = vec![];
 
-    let fields_select = translate(
+    let mut fields_select = translate(
         env,
         state,
         fields,
         &nested_field_table_reference,
         nested_field_from,
+        &mut join_relationship_fields,
```

Comment:
we handle and add joins here, instead in the parent.

Reply:
I like this change.